src/meta/src/hummock/compaction/picker/base_level_compaction_picker.rs

```diff
@@ -99,6 +110,128 @@ impl LevelCompactionPicker {
         }
     }
 
+    fn pick_multi_level_to_base(
+        &self,
+        l0: &OverlappingLevel,
+        target_level: &Level,
+        level_handlers: &[LevelHandler],
+        stats: &mut LocalPickerStatistic,
+    ) -> Option<CompactionInput> {
+        if level_handlers[target_level.level_idx as usize]
+            .is_level_all_pending_compact(target_level)
+        {
+            return None;
```

Comment:
Why? Even if all SSTs in target level are at pending compact status, a trivial move is still possible.

Reply:
Agree

Reply:
good catch ，i will add ut for this case